src/tema4/Main.java

```diff
@@ -0,0 +1,360 @@
+                    System.out.println("The quantity provided is not available, the current stock for the "
+                            + listOfVegetablesProducts.get(selection-1).getId()+ " is " +listOfVegetablesProducts.get(selection-1).getStock() );
+                }
+            }
+        }else{
+            System.out.println("The selection provided was not valid!");
+        }
+    }
+
+    /**
+     * prints out the products from the list listOfSoldItems that have the sellDate attribute as the param
+     * @param date holds a LocalDate value
+     */
+    private static void returnProductsOfSellList(String date){
+        int i = 1;
+        for(LocalStore obj : listOfSoldItems) {
+            if (obj.getSellDate().equals(date)){
+                System.out.println( i + ". " + obj.getId() + obj.getStock());
+                i++;
+            }
+        }
+        if (i == 1){
+            System.out.println("No products were sold in that period");
+        }
+    }
+
+    /**
+     * created and object of class LocalStore with attributes id, stock and sellDate and adds it to the list listOfSoldItems
+     * @param id
+     * @param stock
+     */
+    private static void addToSellList(String id, int stock){
+        LocalStore sellProduct = new LocalStore();
+        sellProduct.setId(id);
+        sellProduct.setSellDate(LocalDate.now().toString());
+        sellProduct.setStock(stock);
+        listOfSoldItems.add(sellProduct);
+    }
+
+    /**
+     * generates the milk and egg objects with atributes and adds them to the list listOfAnimalProducts
+     */
+    private static void mainProducts(){
+        Animal milk = new Animal();
+        milk.setId("milk");
+        milk.setStock(300);
+        milk.setWeight(300);
+        milk.setValidityDate("2019-02-25");
+        milk.setPrice(20);
+        milk.setStorageTemperature(10);
+        Animal egg = new Animal();
+        egg.setId("milk");
+        egg.setStock(300);
+        egg.setWeight(300);
+        egg.setValidityDate("2019-02-25");
+        egg.setPrice(15);
+        egg.setStorageTemperature(15);
+        listOfAnimalProducts.add(milk);
+        listOfAnimalProducts.add(egg);
+
+    }
+
+    /**
+     * diplays all the products from the listOfAnimalProducts and listOfVegetablesProducts
+     */
+    private static void displayTheProducts(){
+        int i = 1;
+        for (LocalStore x : listOfAnimalProducts){
+            System.out.println(i + ". " + x.getId() + " stock: " + x.getStock());
+            i++;
+        }
+        for (LocalStore x : listOfVegetablesProducts){
+            System.out.println(i + ". " + x.getId()+ " stock: " + x.getStock());
+            i++;
+        }
+
+    }
+
+    /**
+     * Menu for the creation of new product
+     */
+    private static void createNewProductMenu(){
+        System.out.println("Product type (1 - animal; 2 - vegetable; 3 - return to previous menu; 4 - exit)");
+        switch (createVariableString("int")){
```

Comment:
a while(variable) until option 4 or until default option where you would change the variable to make the while stop would have been better.
In this way you risk to reach stackOverflow.